esphome/components/max6956/__init__.py

```diff
@@ -0,0 +1,144 @@
+
+
+max6956_ns = cg.esphome_ns.namespace("max6956")
+
+MAX6956 = max6956_ns.class_("MAX6956", cg.Component, i2c.I2CDevice)
+MAX6956GPIOPin = max6956_ns.class_("MAX6956GPIOPin", cg.GPIOPin)
+
+# Actions
+SetCurrentGlobalAction = max6956_ns.class_("SetCurrentGlobalAction", automation.Action)
+SetCurrentModeAction = max6956_ns.class_("SetCurrentModeAction", automation.Action)
+
+MAX6956_CURRENTMODE = max6956_ns.enum("MAX6956CURRENTMODE")
+CURRENT_MODES = {
+    "global": MAX6956_CURRENTMODE.GLOBAL,
+    "segment": MAX6956_CURRENTMODE.SEGMENT,
+}
+
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.declare_id(MAX6956),
+            cv.Optional(CONF_BRIGHTNESS_GLOBAL, default="6%"): cv.percentage,
+            cv.Optional(CONF_BRIGHTNESS_MODE, default="global"): cv.enum(
+                CURRENT_MODES, lower=True
+            ),
+        }
+    )
+    .extend(cv.COMPONENT_SCHEMA)
+    .extend(i2c.i2c_device_schema(0x40))
+)
+
+
+@coroutine
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await i2c.register_i2c_device(var, config)
+    cg.add(var.set_brightness_mode(config[CONF_BRIGHTNESS_MODE]))
+    cg.add(var.set_brightness_global(config[CONF_BRIGHTNESS_GLOBAL]))
+
+
+def validate_mode(value):
+    if not (value[CONF_INPUT] or value[CONF_OUTPUT]):
+        raise cv.Invalid("Mode must be either input or output")
+    if value[CONF_INPUT] and value[CONF_OUTPUT]:
+        raise cv.Invalid("Mode must be either input or output")
+    if value[CONF_PULLUP] and not value[CONF_INPUT]:
+        raise cv.Invalid("Pullup only available with input")
+    return value
+
+
+CONF_MAX6956 = "max6956"
+
+MAX6956_PIN_SCHEMA = cv.All(
+    {
+        cv.GenerateID(): cv.declare_id(MAX6956GPIOPin),
+        cv.Required(CONF_MAX6956): cv.use_id(MAX6956),
+        cv.Required(CONF_NUMBER): cv.int_range(min=4, max=31),
+        cv.Optional(CONF_MODE, default={}): cv.All(
+            {
+                cv.Optional(CONF_INPUT, default=False): cv.boolean,
+                cv.Optional(CONF_PULLUP, default=False): cv.boolean,
+                cv.Optional(CONF_OUTPUT, default=False): cv.boolean,
+            },
+            validate_mode,
+        ),
+        cv.Optional(CONF_INVERTED, default=False): cv.boolean,
+    }
+)
+
+
+@pins.PIN_SCHEMA_REGISTRY.register(CONF_MAX6956, MAX6956_PIN_SCHEMA)
+async def max6956_pin_to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    parent = await cg.get_variable(config[CONF_MAX6956])
+
+    cg.add(var.set_parent(parent))
+
+    num = config[CONF_NUMBER]
+    cg.add(var.set_pin(num))
+    cg.add(var.set_inverted(config[CONF_INVERTED]))
+    cg.add(var.set_flags(pins.gpio_flags_expr(config[CONF_MODE])))
+    return var
+
+
+@automation.register_action(
+    "max6956.set_brightness_global",
+    SetCurrentGlobalAction,
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.use_id(MAX6956),
```

Comment:
You can combine this with `cv.maybe_simple_value` to make the config smaller for users:
```yaml
- max6956.set_brightness_global: 10%
```
Do see my comment above about this not being a percentage though...

```suggestion
            cv.GenerateID(CONF_ID): cv.use_id(MAX6956),
```